cli/flox/src/commands/envs.rs

```diff
@@ -0,0 +1,204 @@
+        Ok(())
+    }
+
+    /// Print all environments
+    ///
+    /// If `--json` is passed, print a JSON object with `active` and `inactive` keys.
+    /// If any environments are active, print them first.
+    /// Then print all inactive environments.
+    /// If no environments are known to Flox, print an appropriate message.
+    fn handle_all(
+        &self,
+        active: ActiveEnvironments,
+        registered: impl Iterator<Item = UninitializedEnvironment>,
+    ) -> Result<()> {
+        let inactive = get_inactive_environments(registered, active.iter())?;
+
+        if self.json {
+            println!(
+                "{:#}",
+                json!({
+                    "active": active,
+                    "inactive": inactive,
+                })
+            );
+            return Ok(());
+        }
+
+        if active.iter().next().is_none() && inactive.is_empty() {
+            message::plain("No environments known to Flox");
+        }
+
+        if active.iter().next().is_some() {
+            message::plain("Active environments:");
+            println!("{}", DisplayEnvironments::new(active.iter(), true));
+        }
+
+        if !inactive.is_empty() {
+            message::plain("Inactive environments:");
+            println!("{}", DisplayEnvironments::new(inactive.iter(), false));
+        }
+
+        Ok(())
+    }
+}
+
+struct DisplayEnvironments<'a> {
+    envs: Vec<&'a UninitializedEnvironment>,
+    format_active: bool,
+}
+
+impl<'a> DisplayEnvironments<'a> {
+    fn new(
+        envs: impl IntoIterator<Item = &'a UninitializedEnvironment>,
+        format_active: bool,
+    ) -> Self {
+        Self {
+            envs: envs.into_iter().collect(),
+            format_active,
+        }
+    }
+}
+
+impl<'a> Display for DisplayEnvironments<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let widest = self
+            .envs
+            .iter()
+            .map(|env| env.name().as_ref().len())
+            .max()
+            .unwrap_or(0);
+
+        let mut envs = self.envs.iter();
+
+        if self.format_active {
+            let Some(first) = envs.next() else {
+                return Ok(());
+            };
+            let first_formatted =
+                format!("{:<widest$}  {}", first.name(), format_path(first.path())).bold();
+            writeln!(f, "{first_formatted}")?;
+        }
+
+        for env in envs {
+            writeln!(f, "{:<widest$}  {}", env.name(), format_path(env.path()))?;
+        }
+
+        Ok(())
+    }
+}
+
+fn format_path(path: Option<&Path>) -> String {
+    path.map(|p| p.display().to_string())
+        .unwrap_or_else(|| "(remote)".to_string())
+}
+
+fn get_registered_environments(
+    registry: &EnvRegistry,
+) -> impl Iterator<Item = UninitializedEnvironment> + '_ {
+    registry.entries.iter().filter_map(|entry| {
+        let path = entry.path.parent()?.to_path_buf();
```

Comment:
@zmitchell the env registry is currently storing the path to the `.flox` directory (including the `.flox` component).
`DotFlox` assumes the path being the parent of `.flox`.
I think we should align both since it becomes rather confusing to tell when we expect the path to contain `.flox` or not.

Reply:
Let's do it

Reply:
broke that out into #1349